packages/plugin-ma-tracking/src/ma-tracking.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AdditionalBrowserOptions,
+  BaseEvent,
+  BeforePlugin,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import type { AmplitudeBrowser } from '@amplitude/analytics-browser';
+import { AdvancedCampaignTracker } from './advanced-campaign-tracker';
+
+export class MATrackingPlugin implements BeforePlugin {
```

Comment:
Im thinking if we should just go functions vs classes. Wanted to update other plugins and examples too

Reply:
It would be great to support both classes and functions.

Reply:
both are supported but i think functions are just more industry standard and better to show as examples to customers